crates/bevy_window/src/window.rs

```diff
@@ -179,10 +264,165 @@ impl WindowResizeConstraints {
+#[derive(Debug, Component, Copy, Clone, Reflect)]
+#[reflect(Component)]
+pub struct Cursor {
+    icon: CursorIcon,
+    visible: bool,
+    grab_mode: CursorGrabMode,
+}
+
+impl Default for Cursor {
+    fn default() -> Self {
+        Cursor {
+            icon: CursorIcon::Default,
+            visible: true,
+            grab_mode: CursorGrabMode::None,
+        }
+    }
+}
+
+impl Cursor {
+    /// Creates a new [`Cursor`].
+    pub fn new(icon: CursorIcon, visible: bool, grab_mode: CursorGrabMode) -> Self {
+        Self {
+            icon,
+            visible,
+            grab_mode,
+        }
+    }
+
+    /// Get the current [`CursorIcon`].
+    #[inline]
+    pub fn icon(&self) -> CursorIcon {
+        self.icon
+    }
+
+    /// Get whether or not the cursor is visible.
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **`Windows`**, **`X11`**, and **`Wayland`**: The cursor is hidden only when inside the window.
+    /// To stop the cursor from leaving the window, use [`set_cursor_lock_mode`](Window::set_cursor_lock_mode).
+    /// - **`macOS`**: The cursor is hidden only when the window is focused.
+    /// - **`iOS`** and **`Android`** do not have cursors
+    #[inline]
+    pub fn visible(&self) -> bool {
+        self.visible
+    }
+
+    /// Get whether or not the cursor is locked.
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **`macOS`** doesn't support cursor lock, but most windowing plugins can emulate it.
+    /// See [issue #4875](https://github.com/bevyengine/bevy/issues/4875#issuecomment-1153977546) for more information.
+    /// - **`iOS/Android`** don't have cursors.
+    #[inline]
+    pub fn grab_mode(&self) -> CursorGrabMode {
+        self.grab_mode
+    }
+
+    /// Set the [`CursorIcon`].
+    pub fn set_icon(&mut self, icon: CursorIcon) {
+        self.icon = icon;
+    }
+
+    /// Set whether or not the cursor is visible.
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **`Windows`**, **`X11`**, and **`Wayland`**: The cursor is hidden only when inside the window.
+    /// To stop the cursor from leaving the window, use [`set_cursor_lock_mode`](Window::set_cursor_lock_mode).
+    /// - **`macOS`**: The cursor is hidden only when the window is focused.
+    /// - **`iOS`** and **`Android`** do not have cursors
+    pub fn set_visible(&mut self, visible: bool) {
+        self.visible = visible;
+    }
+
+    /// Set whether or not the cursor is locked.
+    ///
+    /// This doesn't hide the cursor. For that, use [`set_cursor_visibility`](Window::set_cursor_visibility)
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **`macOS`** doesn't support cursor lock, but most windowing plugins can emulate it.
+    /// See [issue #4875](https://github.com/bevyengine/bevy/issues/4875#issuecomment-1153977546) for more information.
+    /// - **`iOS/Android`** don't have cursors.
+    pub fn set_grab_mode(&mut self, grab_mode: CursorGrabMode) {
```

Comment:
Double-check that we don't get a regression of #6599 with this code.

Reply:
Should be good here now